### ISSUE TITLE
Ability to flip disk

### ISSUE THREAD

**Issue by bert003** [ColinPitrat/caprice32#21]
Is it possible to include the option to flip a disk?

I recently downloaded Untouchables 2 and at one point the game prompted to flip the disk to side B.

Early emulators like CPCE have this option. Was wondering whether it is possible to include it (if not already there and I overlooked it) in caprice32?

**Comment by ColinPitrat** [comment]
Sounds like a good idea.

The current way to do it is through the menu (F1 -> Load/Save and select the second disk) but I can get that it's a bit heavy (especially if you have to do it multiple times). Moreover I noticed recently the menu doesn't support zip files and is broken when passing a zip file on the command line (which reminds me to open a bug for it :-)).

How would you see this feature ?

**Comment by bert003** [comment]
As I imagine it, it could be assigned to shortcut key. At first you put Disk 1 and Disk 2 in Drive A and Drive B respectively and then in the game, when asked, you press the shortcut key and the Disk 2 becomes the active disk.

It could also be an added option in the menu, just in case one forgets the shortcut key/s.

At this point I feel that some documentation is needed for this great emulator project. I could also include it in my live system.

I am planning to complete it once caprice32 is as feature rich as possible. I will continue submitting any issues I find while using the emulator to enhance it as much as possible.

**Comment by ColinPitrat** [comment]
So, current behavior is that all files on the command line are considered in ordered to fill drive A, then drive B, then snapshot, then tape and finally cartridge. The only case in which a file can match 2 cases is for drive A and drive B which both accept the same format.

For zip, there's an assumption that only one type of file will be present in it. If it's not the case, the first match above will be applied (so if there's a .dsk and a .voc the zip will go in the drive A, unless both drive A and drive B were already filled).

If 3 dsk files are provided, at the moment the first one goes in driveA, the second in driveB and the third is ignored.

Now, there are multiple options to support multiple files per drive:
 - revisit completely the way this is handled, introducing flags to specify in which drive a given file goes
 - put everything in drive A unless some flag is specified (e.g. --driveB)
 - keep the current logic but if the file is a zip and contains multiple files, put all the files in the drive instead of only the first one

For backward compatibility and because I think this covers all the actual needs, I think I prefer the last option.

Sketch of what this would look like:
 - `CPC.drvA_file` is changed to be a `std::deque` - it will have a single element when called with a non-zip file but when called with a zip, it will have all the files of disk type. The same could be done for `drvB_file` and potentially `tape_file` although this seems less useful. It seems completely useless to do this for `snap_file` or `cart_file`
 - Calls to `file_load` with `drvA_file` are updated to call it with `drvA_file[0]`
 - At load time, an OSD message is displayed saying that multiple media are in driveA and that they can be switched with a new shortcut (e.g. `CAP32_SWITCHDISKA` == Shift + F5). If we want to support driveB (and tape) we can use Shift + F6 (and Shift + F7)
 - When using the shortcut, we simply rotate filenames in `drvA_file` (`pop_front` / `push_back`) and an OSD message is displayed showing the name of the new `drvA_file[0]`
 - Ideally, this would also be controlable from the UI but this is less critical

**Comment by ColinPitrat** [comment]
Nope, this approach doesn't work, `CPC.drvA_file` is the zip file at the moment.

Instead, let's introduce a `CPC.drvA_zip_index` (and same for drvB and tape).

**Comment by ColinPitrat** [comment]
Some basic support is now available for drive A: if a zip containing multiple .dsk (or .ipf, or .raw, but they must all have the same type) then it's possible to switch between them with Shift+F5.

**Comment by ColinPitrat** [comment]
Considering this as implemented at this point.
